src/Microsoft.AspNetCore.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -343,6 +329,110 @@ private string DebuggerToString()
             }
         }
 
+        /// <summary>
+        /// Returns the <see cref="IHtmlContent"/> body.
+        /// </summary>
+        /// <returns>The inner <see cref="IHtmlContent"/>.</returns>
+        public IHtmlContent RenderBody()
+        {
+            return _innerHtml;
+        }
+
+        /// <summary>
+        /// Renders the start tag for the given <see cref="TagBuilder"/>.
+        /// </summary>
+        /// <returns>The start tag for the <see cref="TagName"/>.</returns>
+        public IHtmlContent RenderStartTag()
+        {
+            return new StartTag(this);
+        }
+
+        private class StartTag : IHtmlContent
```

Comment:
Please also move the new private class or classes down to the bottom of the file. Don't mix them into the methods.